File: day2/js/day1-part1.js
import { input } from "./input.js";

function parseInput(input) {
  let arr = [];
  arr = input.split(`\n`);
  let newArr = arr.map((x) => {
    let y = x.split(" ").map(Number);
    return y;
  });

  return newArr;
}

function main() {
  let inputData = parseInput(input);

  let sum = 0;
  for (let i = 0; i < inputData.length; i++) {
    let increasing = true;
    let decreasing = true;

    let temp = inputData[i];

    for (let x = 1; x < temp.length; x++) {
      if (temp[x] < temp[x - 1]) {
        increasing = false;
      } else {
        decreasing = false;
      }
    }

    if (increasing || decreasing) {
      let isSafe = true;

      // Check the difference between adjacent levels
      for (let j = 1; j < temp.length; j++) {
        let diff = Math.abs(temp[j] - temp[j - 1]);
        if (diff < 1 || diff > 3) {
          isSafe = false;
          break;
        }
      }

      if (isSafe) {
        sum += 1;
      }
    }
  }

  console.log(sum);
}

main();

File: day2/go/day1-part1.go
package main

import (
	"fmt"
	"sort"
	"strconv"
	"strings"
)

func parseInput(s string, leftArr *[]int, rightArr *[]int){
	lines := strings.Split(s, "\n")
	// fmt.Println((lines))

	for i:=0; i<len(lines); i++{
		temp := strings.Split(lines[i], "   ")
		left,_ := strconv.Atoi(temp[0])
		right,_ := strconv.Atoi(temp[1])
		if len(temp) == 2{
			*leftArr = append(*leftArr, left)
			*rightArr = append(*rightArr, right)
		}

	}

}

func main(){
	leftArr := []int{}
	rightArr := []int{}
	parseInput(inputData, &leftArr, &rightArr)

	sort.Ints(leftArr)
	sort.Ints(rightArr)

	// fmt.Println(leftArr, rightArr)
	sum := 0
	for i:=0 ; i<len(leftArr); i++{
		l := leftArr[i]
		r := rightArr[i]

		if (l>r){
			sum+= l-r
		}else{
			sum+= r-l
		}

	}

	fmt.Println(sum)

}

